assets/src/domain/eventEditor/ui/ticketAssignmentsManager/types.ts

```diff
@@ -0,0 +1,58 @@
+import { EntityId } from '@appServices/apollo/types';
+import { RelationsManager } from '@appServices/apollo/relations';
+import { Datetime, Ticket } from '@edtrServices/apollo/types';
+
+export type AssignmentType = 'forDate' | 'forTicket' | 'forAll';
+
+export type AssignmentStatus = 'OLD' | 'NEW' | 'REMOVED';
+
+export interface TAMProps {
+	assignmentType: AssignmentType;
+	entityId?: EntityId;
+}
+
+export interface AssignmentFnArgs {
+	datetimeId: EntityId;
+	ticketId: EntityId;
+}
+
+export interface SetAssignmentFnArgs extends AssignmentFnArgs {
+	remove?: boolean;
+}
+
+export interface AssignmentManager {
+	getData: RelationsManager['getData'];
+	initialize: RelationsManager['setData'];
```

Comment:
MORE VOODOO !!!

what is this called?
I looked in the TS docs but didn't see anything like this!

Reply:
You can get the type of a property of a `type` or `interface` using `Object[key]` syntax.
e.g.
```ts
interface Person {
    name: string;
    age: number;
}

const newAge: Person['age'] = 40;
```

**Note:** Dot notation is used for namespaces
```ts
namespace Espresso {
    export interface Datetime {
        id: string;
        name: string;
    }
}

const date: Espresso.Datetime = {...}
```